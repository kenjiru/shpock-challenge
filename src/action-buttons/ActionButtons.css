.action-buttons {
    text-align: center;
}

.action-buttons > div {
    margin: 0 5px;
}
