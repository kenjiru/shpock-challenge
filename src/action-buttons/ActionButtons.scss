.action-buttons {
    text-align: center;

    > div {
        margin: 0 5px;
    }
}
