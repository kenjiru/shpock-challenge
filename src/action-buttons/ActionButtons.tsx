import * as React from "react";
import { Component, ReactElement } from "react";
import { RaisedButton } from "material-ui";

import "./ActionButtons.css";

class ActionButtons extends Component<IActionButtonsProps> {
    public render(): ReactElement<HTMLElement> {
        return (
            <div className="action-buttons">
                <RaisedButton
                    className="reset"
                    label="Clear Filters"
                    secondary={true}
                    onClick={this.props.onReset}
                />

                <RaisedButton
                    className="submit"
                    label="Search"
                    primary={true}
                    onClick={this.props.onSubmit}
                    disabled={this.props.isSubmitDisabled}
                />
            </div>
        );
    }
}

interface IActionButtonsProps {
    onReset: () => void;
    onSubmit: () => void;
    isSubmitDisabled: boolean;
}

export default ActionButtons;
