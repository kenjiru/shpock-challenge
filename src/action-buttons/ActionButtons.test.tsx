import * as React from "react";
import { RaisedButton } from "material-ui";
import { shallow, ShallowWrapper } from "enzyme";
import shallowToJson from "enzyme-to-json";

import ActionButtons from "./ActionButtons";

describe("ActionButtons", () => {
    it("should render correctly", () => {
        const fn = jest.fn();

        const output = shallow(
            <ActionButtons isSubmitDisabled={false} onSubmit={fn} onReset={fn}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("should have the submit button disabled if the isSubmitDisabled prop is true", () => {
        const fn = jest.fn();
        const output: ShallowWrapper = shallow(
            <ActionButtons isSubmitDisabled={true} onSubmit={fn} onReset={fn}/>
        );

        expect(output.find(RaisedButton).length).toEqual(2);

        expect(output.containsMatchingElement(<RaisedButton
                label="Search"
                primary={true}
                disabled={true}/>
            )
        ).toBe(true);
    });

    it("should handle clicking on the reset button", () => {
        const fn = jest.fn();
        const handleReset = jest.fn();
        const output: ShallowWrapper = shallow(
            <ActionButtons isSubmitDisabled={true} onSubmit={fn} onReset={handleReset}/>
        );

        output.find(".reset").simulate("click");

        expect(handleReset).toHaveBeenCalled();
    });

    it("should handle clicking on the submit button", () => {
        const fn = jest.fn();
        const handleSubmit = jest.fn();
        const output: ShallowWrapper = shallow(
            <ActionButtons isSubmitDisabled={true} onSubmit={handleSubmit} onReset={fn}/>
        );

        output.find(".submit").simulate("click");

        expect(handleSubmit).toHaveBeenCalled();
    });

});
