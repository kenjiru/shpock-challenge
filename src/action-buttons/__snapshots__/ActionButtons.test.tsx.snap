// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButtons should render correctly 1`] = `
<div
  className="action-buttons"
>
  <RaisedButton
    className="reset"
    disabled={false}
    fullWidth={false}
    label="Clear Filters"
    labelPosition="after"
    onClick={[Function]}
    primary={false}
    secondary={true}
  />
  <RaisedButton
    className="submit"
    disabled={false}
    fullWidth={false}
    label="Search"
    labelPosition="after"
    onClick={[Function]}
    primary={true}
    secondary={false}
  />
</div>
`;
